pkg/cluster/gitrepo/gitrepo_cluster.go

```diff
@@ -997,58 +1068,89 @@ func (g *clusterGitRepo) Rollback(ctx context.Context, application, cluster, com
 
 	pid := fmt.Sprintf("%v/%v/%v", g.clustersGroup.FullPath, application, cluster)
 
-	// 1. get cluster files of the specified commit
-	// the files contains: GitopsFilePipeline, GitopsFileApplication, GitopsFileBase, GitopsFilePipelineOutput
-	var err1, err2, err3, err4, err5, err6 error
-	var applicationBytes, pipelineBytes, baseValueBytes, pipelineOutPutBytes, tagsBytes, chartBytes []byte
-	var wgReadFile sync.WaitGroup
-	wgReadFile.Add(6)
-	readFile := func(b *[]byte, err *error, filePath string) {
-		defer wgReadFile.Done()
-		bytes, e := g.gitlabLib.GetFile(ctx, pid, commit, filePath)
-		*b = bytes
-		*err = e
-	}
-	go readFile(&pipelineBytes, &err1, common.GitopsFilePipeline)
-	go readFile(&applicationBytes, &err2, common.GitopsFileApplication)
-	go readFile(&baseValueBytes, &err3, common.GitopsFileBase)
-	go readFile(&pipelineOutPutBytes, &err4, common.GitopsFilePipelineOutput)
-	go readFile(&tagsBytes, &err5, common.GitopsFileTags)
-	go readFile(&chartBytes, &err6, common.GitopsFileChart)
-	wgReadFile.Wait()
+	// compare commit straight diffs
+	compare, err := g.gitlabLib.Compare(ctx, pid, GitOpsBranch, commit, utilcommon.BoolPtr(true))
+	if err != nil {
+		return "", err
+	}
+	if compare.Diffs == nil {
+		return "", perror.Wrapf(herrors.ErrParamInvalid,
+			"commit dose not support rollback, commit = %s", commit)
+	}
 
-	for _, err := range []error{err1, err2, err3, err4, err6} {
+	readFile := func(fileName string) ([]byte, error) {
+		return g.gitlabLib.GetFile(ctx, pid, commit, fileName)
+	}
+
+	var wg sync.WaitGroup
+
+	type actionCase struct {
+		diff   gitlab.Diff
+		action gitlablib.CommitAction
+		err    error
+	}
+
+	revertAction := func(oneCase *actionCase) {
+		defer wg.Done()
+
```

Comment:
add common， and a simple complete  example。

do not use lamda

Reply:
fixed